app/scripts/directives/serviceIntegration.js

```diff
@@ -0,0 +1,241 @@
+      var version = {
+        group:"settings.k8s.io",
+        resource:"podpresets",
+        version:"v1alpha1"
+      };
+
+      var isBindingReady = $filter("isBindingReady");
+
+      //binding might not be ready currently, so
+      //watch binding to wait for it to be ready
+      var bindingReadyWatch = DataService.watch(APIService.getPreferredVersion("servicebindings"), context, function(bindings){
+        _.each(bindings._data, function(iterateBinding){
+          if(iterateBinding.metadata.name === binding.metadata.name && isBindingReady(iterateBinding)){
+            DataService.unwatch(bindingReadyWatch);
+            _.set(iterateBinding, "metadata.annotations.consumer", ctrl.consumerService.metadata.labels.serviceName);
+            _.set(iterateBinding, "metadata.annotations.provider", ctrl.integration);
+            DataService.update(APIService.getPreferredVersion("servicebindings"), iterateBinding.metadata.name, iterateBinding, context)
+          }
+        });
+      });
+
+
+      DataService.create(version, null, podPreset, context)
+      .then(function() {
+        return DataService.get(APIService.getPreferredVersion('deployments'), _.get(ctrl.consumerService, 'metadata.labels.serviceName'), context, {
+          errorNotification: false
+        });
+      })
+      .then(function(deployment) {
+        deployment.spec.template.metadata.labels[_.get(ctrl.serviceInstance, 'metadata.labels.serviceName')] = "enabled";
+        return DataService.update(APIService.getPreferredVersion('deployments'), _.get(ctrl, 'consumerService.metadata.labels.serviceName'), deployment, context);
+      })
+      .then(function() {
+        NotificationsService.addNotification({
+          type: "success",
+          message: "Binding has been set in " + _.get(ctrl, 'consumerService.metadata.labels.serviceName') + " Redeployed " + _.get(ctrl, 'consumerService.metadata.labels.serviceName')
+        });
+      })
+      .catch(function(err) {
+        NotificationsService.addNotification({
+          type: "error",
+          message: "Failed to integrate service binding",
+          details: err.data.message
+        });
+      });
+    };
+
+    ctrl.getState = function(){
+      if(ctrl.podPreset && !ctrl.binding) {
+        return 2;
+      }
+      if(ctrl.podPreset && ctrl.binding) {
+        return 1;
+      }
+      if(ctrl.binding) {
+        var state=1;
+        _.each(ctrl.binding.status.conditions, function(condition){
+          if(condition.type === "Ready" && condition.status !== "True"){
+            state=2;
+          }
+        });
+        return state;
+      }
+      if (ctrl.serviceInstance){
+        return 0;
+      }
+      return -1;
+    };
+
+    //called in callback from succesful delete-link for binding
+    ctrl.deletePodPreset = function(){
+      var context = {namespace: ctrl.consumerService.metadata.namespace};
+      var deleteOptions = {propagationPolicy: null};
```

Comment:
Just a question about what this does?

Reply:
it's required to make the binding delete immediately, otherwise it seems to be deleted with very low priority and take forever to go.

Reply:
:+1: